### PR TITLE
implemented DetoneCovarianceTransformer

### PR DESCRIPTION
**Pull request by moneygeek**

### CODE REVIEW

File: mcos/covariance_transformer.py [hunk 1]
```diff
@@ -177,24 +207,48 @@ def _de_noised_corr(self, eigenvalues: np.array, eigenvectors: np.array, n_facts
...
-    corr[corr < -1], corr[corr > 1] = -1, 1  # numerical error
-    return corr
+        corr = cov_to_corr(cov)
+
+        w, v = linalg.eig(corr)
+
+        # sort from highest eigenvalues to lowest
+        sort_index = np.argsort(-np.abs(w))  # get sort_index in descending absolute order - i.e. from most significant
+        w = w[sort_index]
+        v = reorder_matrix(v, sort_index)
```
Comment (stevey628) [inline mcos/covariance_transformer.py:233]: reorder_matrix reorders both columns and rows of the matrix. won't reordering the rows of the eigenvectors change the vectors themselves?

**Comment by moneygeek** [comment]
It does, and it should. By the way, we get the reordered corr matrix if we
use the recorded w and v to do v*diag(w)*v'

**Comment by moneygeek** [comment]
*reordered

**Comment by moneygeek** [comment]
actually, I guess I wouldn't have to reorder the corr matrix if I only
swapped the eigenvectors around. I'll update the code.

**Pull request merged**